pkg/ibm_plugin/sdk/resource.go

```diff
@@ -0,0 +1,513 @@
+	// A Cluster would have a security group with prefix of 'kube-',
+	// We infer the VPC of the cluster using the VPC of this security group
+	clusterSG := "kube-" + c.ID
+	sgs, _, err := c.client.vpcService.ListSecurityGroups(c.client.vpcService.NewListSecurityGroupsOptions())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, sg := range sgs.SecurityGroups {
+		if *sg.Name == clusterSG {
+			return sg.VPC, nil
+		}
+	}
+	return nil, fmt.Errorf("unable to find the VPC of cluster %s", c.ID)
+}
+
+// NOTE: Currently not in use, as public ips are not provisioned.
+// deletes floating ips marked recyclable, that are attached to
+// any interface associated with given VM
+func (c *CloudClient) deleteFloatingIPsOfVM(vm *vpcv1.Instance) {
+	recyclableResource := "recyclable" // placeholder indicator
+
+	for _, nic := range vm.NetworkInterfaces {
+		options := c.vpcService.NewListInstanceNetworkInterfaceFloatingIpsOptions(*vm.ID, *nic.ID)
+		ips, _, err := c.vpcService.ListInstanceNetworkInterfaceFloatingIps(options)
+		if err != nil {
+			utils.Log.Println(err)
+		}
+		for _, ip := range ips.FloatingIps {
+			if strings.Contains(recyclableResource, *ip.Name) {
+				_, err := c.vpcService.DeleteFloatingIP(c.vpcService.NewDeleteFloatingIPOptions(*ip.ID))
+				if err != nil {
+					utils.Log.Println(err)
+				}
+				utils.Log.Println("Deleted recyclable IP: ", *ip.Address)
+			}
+		}
+	}
+}
+
+// returns true when instance is completely removed from the subnet.
+func (c *CloudClient) waitForInstanceRemoval(vmID string) bool {
+	sleepDuration := 10 * time.Second
+	for tries := 15; tries > 0; tries-- {
+		_, _, err := c.vpcService.GetInstance(c.vpcService.NewGetInstanceOptions(vmID))
+		if err != nil {
+			return true
+		}
+		time.Sleep(sleepDuration)
+	}
+	return false
+}
+
+// GetResourceHandlerFromDesc gets the resource handler from the resource description
+func (c *CloudClient) GetResourceHandlerFromDesc(resourceDesc []byte) (ResourceIntf, error) {
+	instanceOptions := vpcv1.CreateInstanceOptions{
+		InstancePrototype: &vpcv1.InstancePrototypeInstanceByImage{
+			Image:   &vpcv1.ImageIdentityByID{},
+			Zone:    &vpcv1.ZoneIdentityByName{},
+			Profile: &vpcv1.InstanceProfileIdentityByName{},
+		},
+	}
+
+	clusterOptions := k8sv1.VpcCreateClusterOptions{}
+
+	err := json.Unmarshal(resourceDesc, &clusterOptions)
+	if err == nil && clusterOptions.WorkerPool != nil {
+		return &ResourceClusterType{client: c}, nil
+	}
+
+	err = json.Unmarshal(resourceDesc, &instanceOptions)
+	if err == nil && instanceOptions.InstancePrototype != nil {
+		return &ResourceInstanceType{client: c}, nil
+	}
+
+	return nil, fmt.Errorf("failed to unmarshal resource description:%+v", err)
+
+}
+
+// GetResourceHandlerFromID gets the resource handler from the resource ID/URI
+func (c *CloudClient) GetResourceHandlerFromID(deploymentID string) (ResourceIntf, error) {
+	parts := strings.Split(deploymentID, "/")
+
+	if len(parts) >= 5 {
```

Comment:
Is this better than checking for exact length of 6?

Reply:
Since I am checking parts[5], dint want to check the exact length of 6, and checking >=5 seems more readable :)